src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.SdkArchiveDiff/GetSingleTarballItem.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+
+public class GetSingleTarballItem : Microsoft.Build.Utilities.Task
```

Comment:
I think maybe the intention is to use this for more, but technically you don't need a task to select a single item from the itemgroup. ItemGroups have intrinsic size functions for checking 1 or 0 cases, and you can use the following to select the first (or last without Reverse). Probably less efficient though.

```
<Target Name="FirstCompile" BeforeTargets="Compile">
  <ItemGroup>
    <Reversed Include="@(Compile->Reverse())" />
  </ItemGroup>
  <PropertyGroup>
    <First>%(Reversed.Identity)</First>
  </PropertyGroup>

  <Message Text="First compile item is $(First)" Importance="high" />
</Target>
```

Reply:
Yeah, this task tests each item to make sure we select an item from the group that the next task can parse the version and RID info from, not just the first item.